third_party/openocd_cmake/CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+cmake_minimum_required(VERSION 3.15)
+
+if (${CMAKE_VERSION} VERSION_GREATER "3.8")
+    #For cmake >= 3.9 INTERPROCEDURAL_OPTIMIZATION behaviour we need to explicitly
+    #set the cmake policy version number
+    cmake_policy(VERSION 3.9) 
+
+    # If we are using verison < 3.9 then setting INTERPROCEDURAL_OPTIMIZATION
+    # has no effect unless an Intel compiler is used
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+    message(WARNING "Build type not set, falling back to Release mode.
+        To specify build type use:
+        -DCMAKE_BUILD_TYPE=<mode> where <mode> is Debug or Release.")
+    set(CMAKE_BUILD_TYPE
+        "Release"
+        CACHE STRING "Choose the type of build, options are: Debug Release."
+        FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
+project(rs_openocd)
+
+if(RAPTOR)
+    set(root_dir ${PROJECT_SOURCE_DIR}/../../../..)
+else()
+    if ((EXISTS ${PROJECT_SOURCE_DIR}/../../build) OR (EXISTS ${PROJECT_SOURCE_DIR}/../../dbuild))
+        set(root_dir ${PROJECT_SOURCE_DIR}/../..)
+    else()
+        set(root_dir ${PROJECT_SOURCE_DIR}/../../..)
+    endif()
+endif()
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(build_dir ${root_dir}/dbuild)
+else ()
+    set(build_dir ${root_dir}/build)
+endif()
+
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+    set(rs_openocd_tar "openocd-i686-w64-mingw32.tar.gz")
+else()
+    set(rs_openocd_tar "openocd-linux86_64.tar.gz")
+endif()
+
+if(RAPTOR)
+    set(Destination_Dir ${build_dir}/share/raptor/configuration)
```

Comment:
As we talked about, it's possible to merge this after line 25. However, this is being done to improve the sequence of information for better readability.